### ISSUE TITLE
No IV left after thresholding

### ISSUE THREAD

**Issue by xz-xxz** [n-mounier/MRlap#21]
Hi, I'm trying to use MRlap to support my findings.  When using function 'TwosampleMR', more than 100 snps remained, but error occurred when running IVW MR using 'MRlap' as "no IV left after thresholding (5e-08). Please advise, thanks.
![屏幕截图 2024-08-02 201919](https://github.com/user-attachments/assets/8dee42a7-8259-46ff-8b31-5b1ed82609d4)

**Comment by yangyizhixiaomaomi** [comment]
Have you found a solution to ldsc that Error in h(simpleError(msg, call)) : 
  在为't'函数选择方法时评估'x'参数出了错: 非整合参数

**Comment by xz-xxz** [comment]
This problem could be solved by using the full GWAS statistics. Have you try? Additionally, I still have this problem, as no IV left after thresholding.

**Comment by yangyizhixiaomaomi** [comment]
yes when I use the whole gwas ，it successd.you can try .for another problem ,l haven't . maybe your p_value too strict? l don't know.



---Original---
From: ***@***.***&gt;
Date: Sun, Aug 25, 2024 13:42 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [n-mounier/MRlap] No IV left after thresholding (Issue #21)




 
This problem could be solved by using the full GWAS statistics. Have you try? Additionally, I still have this problem, as no IV left after thresholding.

**Comment by xz-xxz** [comment]
@lcpilling  Hi. I still have this problem. "TwosampleMR" identified more than 100 SNPs using threshold of 5e-08 and clumping of 10,000kb (r2=0.001), but "MR lap" showed no IV left when using the same parameters. Additionally, each run costs about 30 mins. Could you please check this, I don't know how to solve this error. 
![屏幕截图 2024-08-25 175827](https://github.com/user-attachments/assets/46930bee-d8d1-4464-a9d8-d50fe2889bf0)

**Comment by lcpilling** [comment]
Hi @xz-xxz - sorry to ask but are you certain the object `LST` contains significant (p<5e-08) variants? Could you check with `table( LST$P < 5e-08 )`

**Comment by xz-xxz** [comment]
@lcpilling  Thanks for your reply. Yes, definitely, a total of 7462 variants remained after thresholding of 5e-08.
![屏幕截图 2024-08-28 162602](https://github.com/user-attachments/assets/802c8d3f-d4ae-4a43-b575-ab936d6e989d)

**Comment by xz-xxz** [comment]
![屏幕截图 2024-08-28 164017](https://github.com/user-attachments/assets/201d56d8-f7c6-4c85-a3ae-a81f55f256d5)

**Comment by xz-xxz** [comment]
A total of 7438 variants have significant z-statistic.
![屏幕截图 2024-08-28 171732](https://github.com/user-attachments/assets/c8189bd1-a137-4f0d-8400-35488cdd0362)

**Comment by lcpilling** [comment]
OK.... I do not know why MRlap finds no IVs - that is unusual!

**Comment by xz-xxz** [comment]
Well… thank you anyway！





徐振
***@***.***

签名由 网易灵犀办公 定制





Original:
From: Luke C. ***@***.***> Date: 2024-08-28 19:18:43 To: ***@***.***> Cc: ***@***.***>, ***@***.***> Subject: Re: [n-mounier/MRlap] No IV left after thresholding (Issue #21)
OK.... I do not know why MRlap finds no IVs - that is unusual!

**Comment by hyjyd** [comment]
same problem！

**Comment by lcpilling** [comment]
Is the problem with the "N" provided? MRlap may exclude SNPs if the N is very small

**Comment by hyjyd** [comment]
"a","t","c","g" in "A1" "A2"  convert to A T C G will be ok

**Comment by chenyang0819** [comment]
Is anyone solved this yet?

**Comment by ssxxmmm** [comment]
I'm also running into this issue, what a headache! The two-sample MR gives significant results, but running MRlap shows the same problem no IV left after thresholding (5e-08)！Is anyone solved this yet?
